packages/google-cloud-translate/src/v2/index.ts

```diff
@@ -14,19 +14,62 @@
 
 import {Service, util, Metadata} from '@google-cloud/common';
 import {promisifyAll} from '@google-cloud/promisify';
-import arrify = require('arrify');
-import * as extend from 'extend';
+import extend from 'extend';
+import * as fs from 'fs';
 import {GoogleAuthOptions} from 'google-auth-library';
-
-// eslint-disable-next-line @typescript-eslint/no-var-requires
-const isHtml = require('is-html');
 import {
   DecorateRequestOptions,
   BodyResponseCallback,
-} from '@google-cloud/common/build/src/util';
+} from '@google-cloud/common/build/src/util.js';
+import htmlTags from 'html-tags';
+import path from 'path';
+import {fileURLToPath} from 'url';
+let dirToUse = '';
+try {
+  dirToUse = __dirname;
```

Comment:
I would pull this logic into a separate utility library, perhaps in the generator or gax so that we don't have to add this in every generated library.

I think if we're creative with how this helper is implemented, we should be able to avoid the `regexMetaInput` replacement.

What I did in yargs was to have a different helper file for ESM vs., CJS, during the build step you can then use the appropriate helper. Such that the generated code only has __dirname for CJS and import.meta.url for ESM.

Reply:
I agree that the regex is fragile; it's meant to be a working prototype, but I do think regex is ultimately the way to go for the __dirname --> import.meta.url problem. Say, for example, we created a helper file that exports a default function called `dirnameOrImportMeta()`, that lives in gax, for example, that looks like this, and is imported in each file that asks for the __dirname:
```
export default function dirnameOrImportMeta() {
    let dirToUse = '';
    try {
      dirToUse = __dirname;
    } catch (e) {
      // eslint-disable-next-line @typescript-eslint/ban-ts-comment
      // @ts-ignore
      const filename = fileURLToPath(import.meta.url);
      dirToUse = path.dirname(filename);
    }
  
    return dirToUse;
  }
```

This option is fragile in that __dirname would actually be pointing to where this particular file lives, not the file that is calling this function. This could lead to wrong paths. Even if we passed the context of the currently executing directory, such as `process.cwd()`, this also wouldn't necessarily be called from the file that is executing, since we wouldn't know what context that particular file is being run in. 

Another option is to just use this logic in each file that needs it:

```
    let dirToUse = '';
    try {
      dirToUse = __dirname;
    } catch (e) {
      // eslint-disable-next-line @typescript-eslint/ban-ts-comment
      // @ts-ignore
      const filename = fileURLToPath(import.meta.url);
      dirToUse = path.dirname(filename);
    }
  
    return dirToUse;
```
However, CJS tests will complain during runtime:

```
TypeError [ERR_INVALID_URL]: Invalid URL
    at __node_internal_captureLargerStackTrace (node:internal/errors:484:5)
    at new NodeError (node:internal/errors:393:5)
    at URL.onParseError (node:internal/url:565:9)
    at new URL (node:internal/url:645:5)
    at fileURLToPath (node:internal/url:1483:12)
```
Therefore, we'd still need to do some regex replacing to remove the filename part in build/cjs (which is what we're doing in this prototype, hence the fragility).

Another option, we could do something like:

```
const dirname = CHANGE_THIS
```
And change it to be what each file needs in the post-compiling step. However, this option wouldn't allow the file to be runnable (say, for example, from Deno), since it would need to be compiled to be executable.

Lastly, there are libraries that do this transition for you - like `[tsup](https://www.npmjs.com/package/tsup)`, according to this [article](https://antfu.me/posts/publish-esm-and-cjs). The downsides of this alternative are that 1) we'd probably need other custom post-compiling done, like esmock --> proxyquire, and 2) we'd have another dependency that we'd have to maintain and probably does not overlap perfectly with what we need it to do.